### PR TITLE
Night time and solar panel popup

### LINKED ISSUE #284: Show a message when placing solar panels in Diggy
As solar panels make no sense, why not show a message the first time? Could also prevent placing them, but that would simply raise questions all the time.

### PR DESCRIPTION
**Pull request by SimonFlapse**
Split the time assignment into a new file NightTime.lua allowing for it to be disabled. (Restore normal day/night cycle)

Also added a popup when a player places a solar panel informing that they are purely cosmetic

Removed the recipe for portable solar panels because they are useless, but the technology is needed.

The popup is shown every time a solar panel is placed. <--- (Up for discussion)

No desyncs with toggle-heavy-mode

closes #284

### CODE REVIEW

File: map_gen/Diggy/Feature/NightTime.lua [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+Foreman's advice: Solar Panels are only useful in crafting
+satellites
+]]
+        )
+    end
+end
+
+local function on_research_finished(event)
+    local force = game.forces.player
```
Comment (iltar) [inline map_gen/Diggy/Feature/NightTime.lua]: You can also do this to keep it scoped:
```suggestion
    local force = event.research.force
```

https://lua-api.factorio.com/latest/events.html#on_research_finished

File: map_gen/Diggy/Feature/NightTime.lua [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+end
+
+function NightTime.on_init()
+    local surface = game.surfaces.nauvis
+    
+    surface.daytime = 0.5
+    surface.freeze_daytime = 1
+end
+
+return NightTime
```
Comment (iltar) [inline map_gen/Diggy/Feature/NightTime.lua]: image

Can you add that?

File: map_gen/Diggy/Feature/NightTime.lua [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+end
+
+local function on_research_finished(event)
+    local force = game.forces.player
+    force.recipes["solar-panel-equipment"].enabled=false
+end
+
+function NightTime.register(config)
+        Event.add(defines.events.on_built_entity, on_built_entity)
+        Event.add(defines.events.on_research_finished, on_research_finished)
```
Comment (iltar) [inline map_gen/Diggy/Feature/NightTime.lua]: Should have 1 indent, not 2

File: map_gen/Diggy/Feature/NightTime.lua (hunk 1 above)
Comment (SimonFlapse) [reply map_gen/Diggy/Feature/NightTime.lua]: Great! I wondered if I could do that, but forgot to check.

**Pull request merged**